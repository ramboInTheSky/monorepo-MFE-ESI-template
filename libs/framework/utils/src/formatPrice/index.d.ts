export declare const formatPrice: (minPrice: number, maxPrice: number, currency: string, locale: string, realm: string) => any;
export default formatPrice;
