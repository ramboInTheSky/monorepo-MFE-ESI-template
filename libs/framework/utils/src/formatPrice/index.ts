import logger from "@monorepo/core-logger"

import config from "./config.json"

const LEFT_ALIGN = "left"
const RIGHT_ALIGN = "right"
const DEFAULT_MIN_DIGITS = 2
const TYPE_CURRENCY = "currency"
const TYPE_LITERAL = "literal"
const TYPE_GROUP = "group"
const TYPE_DECIMAL = "decimal"

if (typeof Intl !== "object") {
    const expectedValue = "node_modules/full-icu"
    if (!process.env.NODE_ICU_DATA || process.env.NODE_ICU_DATA !== expectedValue) {
        throw new Error(
            `NODE_ICU_DATA environment variable is not set, or not set to the correct value of ${expectedValue}`,
        )
    }
    const err = new Error("Intl is not available. Please install the full-icu module.")
    logger.error(err)
    throw err
}

export const formatPrice = (
    minPrice: number,
    maxPrice: number,
    currency: string,
    locale: string,
    realm: string,
) => {
    return minPrice === maxPrice
        ? formatNumber(minPrice, currency, locale,  realm)
        : `${formatNumber(minPrice, currency, locale,  realm)} - ${formatNumber(
              maxPrice,
              currency,
              locale,
              realm,
          )}`
}

const getCurrencyConfig = (realm: string, locale: string) => {
    const splitLocale = locale?.toLocaleLowerCase().split("-")
    const territory = splitLocale[1]
    const language = splitLocale[0]

    const getId = config.realms[realm]?.[territory]?.[language]?.currencies?.id
    return config.currencies[getId]
}

type ConfigCurrency = {
    currencySymbol: string
    spaceRequired: boolean
    position: string
    decimalSeparatorSymbol: string | null
    groupSeparatorSymbol: string | null
    showNumOfDecimalDigits: number | null
}

const getConfiguredCurrency = (price, locale, configuredCurrency) => {
    const {
        currencySymbol,
        spaceRequired,
        position,
        showNumOfDecimalDigits,
        decimalSeparatorSymbol,
        groupSeparatorSymbol,
    }: ConfigCurrency = configuredCurrency
    let formattedPrice

    const requiredSpacing = spaceRequired ? " " : ""

    
    // showDecimalDigits is used for header mini bag for now
    const numberFormatOptions : Intl.NumberFormatOptions = {
        style: "decimal",
        minimumFractionDigits: showNumOfDecimalDigits ?? DEFAULT_MIN_DIGITS,
        useGrouping: true,
    }
    if (price % 1 === 0) {
        numberFormatOptions.minimumFractionDigits = 0
        numberFormatOptions.maximumFractionDigits = 0
    }

    const configPrice = new Intl.NumberFormat(locale, numberFormatOptions)
    if (configPrice.resolvedOptions().numberingSystem === "latn") {
        let displayedPrice = configPrice.format(price)
        // replacing the group or decimal of the price
        if (groupSeparatorSymbol || decimalSeparatorSymbol) {
            displayedPrice = configPrice
                .formatToParts(price)
                .map(({type, value}) => {
                    if (groupSeparatorSymbol && type === TYPE_GROUP) {
                        // eslint-disable-next-line no-param-reassign
                        value = groupSeparatorSymbol
                    }

                    if (decimalSeparatorSymbol && type === TYPE_DECIMAL) {
                        // eslint-disable-next-line no-param-reassign
                        value = decimalSeparatorSymbol
                    }

                    return value
                })
                .reduce((string, part) => `${string}${part}`)
        }

        switch (position) {
            case LEFT_ALIGN:
                formattedPrice = `${currencySymbol}${requiredSpacing}${displayedPrice}`
                break
            case RIGHT_ALIGN:
                formattedPrice = `${displayedPrice}${requiredSpacing}${currencySymbol}`
                break
            default:
                formattedPrice = `${currencySymbol}${requiredSpacing}${displayedPrice}`
                break
        }
    } else {
        // Logic here is because we need the currency in latin numbers not say arabic, which by default Intl.NumberFormat will convert to
        // If you format sa-AR currency to latin characters the currency symbol is added to wrong side
        // Need to combine correctly formatted numbers, with correctly formatted currency
        const latinCharacterNumberFormat = new Intl.NumberFormat(
            `${locale}-u-nu-latn`,
            numberFormatOptions,
        ).formatToParts(price)

        const numberOnlyLatinCharacterPrice = latinCharacterNumberFormat
            .map(part => {
                if (part.type !== TYPE_CURRENCY && part.type !== TYPE_LITERAL) return part.value
                return ""
            })
            .join("")

        switch (position) {
            case LEFT_ALIGN:
                formattedPrice = `${currencySymbol}${requiredSpacing}${numberOnlyLatinCharacterPrice}`
                break
            case RIGHT_ALIGN:
                formattedPrice = `${numberOnlyLatinCharacterPrice}${requiredSpacing}${currencySymbol}`
                break
            default:
                formattedPrice = `${currencySymbol}${requiredSpacing}${numberOnlyLatinCharacterPrice}`
                break
        }
    }
    return formattedPrice
}

const getCurrency = (price, currency, locale) => {
    // Below is used for other countries apart from apart from batch one countries
    // Please see config.json to see which countries are configured
    // data from config.json is coming from MVC language config

    const numberFormatOptions: Intl.NumberFormatOptions = {
        style: "currency",
        currency,
    }
    if (price % 1 === 0) {
        numberFormatOptions.minimumFractionDigits = 0
        numberFormatOptions.maximumFractionDigits = 0
    }
    const localeNumberFormat = new Intl.NumberFormat(`${locale}`, numberFormatOptions)
    if (localeNumberFormat.resolvedOptions().numberingSystem === "latn") return localeNumberFormat.format(price)

    // Logic here is because we need the currency in latin numbers not say arabic, which by default Intl.NumberFormat will convert to
    // If you format sa-AR currency to latin characters the currency symbol is added to wrong side
    // Need to combine correctly formatted numbers, with correctly formatted currency
    const latinCharacterNumberFormat = new Intl.NumberFormat(`${locale}-u-nu-latn`, numberFormatOptions)
    if (typeof latinCharacterNumberFormat.formatToParts !== "function") return latinCharacterNumberFormat.format(price)

    const latinCharacterNumber = latinCharacterNumberFormat.formatToParts(price)

    const numberOnlyLatinCharacterPrice = latinCharacterNumber
        .map(part => {
            if (part.type !== TYPE_CURRENCY && part.type !== TYPE_LITERAL) return part.value
            return ""
        })
        .join("")

    const currencyPattern = new Intl.NumberFormat(locale, {
        style: "currency",
        currency,
        minimumFractionDigits: 0,
        maximumFractionDigits: 0,
    }).formatToParts(1)

    let formattedPrice = ""
    currencyPattern.forEach(part => {
        if (part.type === "currency") {
            if (part.value.endsWith(".")) {
                formattedPrice += part.value.substring(0, part.value.length - 1)
            } else {
                formattedPrice += part.value
            }
        } else if (part.type === "literal") {
            formattedPrice += part.value
        } else {
            formattedPrice += numberOnlyLatinCharacterPrice
        }
    })

    return formattedPrice
}

const formatNumber = (price: number, currency: string, locale: string, realm: string) => {
    const configuredCurrency = getCurrencyConfig(realm, locale)
    if (configuredCurrency) return getConfiguredCurrency(price, locale, configuredCurrency)
    return getCurrency(price, currency, locale)
}
export default formatPrice
