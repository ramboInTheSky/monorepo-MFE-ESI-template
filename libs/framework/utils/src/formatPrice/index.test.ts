import formatPrice from "."

// import config from "./config.json"
const batchOneTests = [
    "en-ar",
    "en-am",
    "de-at",
    "en-at",
    "en-az",
    "en-by",
    "en-br",
    "bg-bg",
    "en-bg",
    "fr-ca",
    "en-ca",
    "en-cl",
    "en-hr",
    "en-cy",
    "en-cz",
    "cs-cz",
    "da-dk",
    "en-dk",
    "ar-eg",
    "en-eg",
    "ru-ee",
    "en-ee",
    "en-fi",
    "fr-fr",
    "en-fr",
    "en-ge",
    "en-gi",
    "el-gr",
    "en-gr",
    "en-hu",
    "hu-hu",
    "en-is",
    "en-in",
    "en-id",
    "it-it",
    "en-it",
    "ru-lv",
    "en-lv",
    "ar-lb",
    "en-lb",
    "ru-lt",
    "en-lt",
    "fr-lu",
    "en-lu",
    "ms-my",
    "en-my",
    "en-mx",
    "en-nz",
    "en-no",
    "en-om",
    "ar-om",
    "en-pk",
    "en-pe",
    "en-ph",
    "en-pt",
    "en-rs",
    "en-sg",
    "sl-si",
    "en-si",
    "en-za",
    "ko-kr",
    "en-kr",
    "es-es",
    "en-es",
    "sv-se",
    "en-se",
    "de-ch",
    "en-ch",
    "zh-tw",
    "en-tw",
    "en-th",
    "en-tr",
]

describe("Given a price formatter", () => {
    describe("When formatting the price with the config", () => {
        batchOneTests.forEach(locale => {
            const lowercaseLocale = locale.toLocaleLowerCase()
            describe(`locale: ${lowercaseLocale}`, () => {
                it(`should render with no range `, () => {
                    const minPrice = 10
                    const maxPrice = 10

                    const defaultCurrencyCode = "GBP"

                    const realm = "amido"

                    expect(
                        formatPrice(minPrice, maxPrice, defaultCurrencyCode, lowercaseLocale, realm),
                    ).toMatchSnapshot()
                })
                it(`should render with range`, () => {
                    const minPrice = 10
                    const maxPrice = 15

                    const defaultCurrencyCode = "GBP"
                    const realm = "amido"

                    expect(
                        formatPrice(minPrice, maxPrice, defaultCurrencyCode, lowercaseLocale, realm),
                    ).toMatchSnapshot()
                })
            })
        })
    })

    describe("When formatting the price", () => {
        it("When there is no range, it should return correctly formatted price", () => {
            const realm = "amido"
            expect(formatPrice(10, 10, "GBP", "en-GB", realm)).toEqual("£10")
        })

        it("When there is no range and decimals, it should return correctly formatted price", () => {
            const realm = "amido"
            expect(formatPrice(10.5, 10.5, "GBP", "en-GB", realm)).toEqual("£10.50")
        })

        it("When there is a range, it should return correctly formatted price", () => {
            const realm = "amido"
            expect(formatPrice(10, 15, "GBP", "en-GB", realm)).toEqual("£10 - £15")
        })

        it("When there is no price, it should return correctly formatted price", () => {
            const realm = "amido"
            expect(formatPrice(0, 0, "GBP", "en-GB", realm)).toEqual("£0")
        })

        it("When in Saudi English, it should return correctly formatted price", () => {
            const realm = "amido"
            expect(formatPrice(10, 15, "SAR", "en-SA", realm)).toBe("SAR 10 - SAR 15")
        })

        it("When in Saudi Arabic, it should return correctly formatted price", () => {
            const realm = "amido"
            expect(formatPrice(10, 10, "SAR", "ar-SA", realm)).toBe("10 ر.س‏")
        })

        // Bug 34157
        it("should show 2 digits for Thailand correctly when there are pennies", () => {
            const realm = "amido"
            expect(formatPrice(10.5, 12, "USD", "en-th", realm)).toEqual("$10.50 - $12")
        })

        // Bug 34175
        it("should show 0 digits for Malaysia when no pennies", () => {
            const realm = "amido"
            expect(formatPrice(10.5, 12, "MYR", "ms-my", realm)).toEqual("MYR 10.50 - MYR 12")
        })

        // Bug 34159
        it("should show 1000 delimiter as . for Switzerland DE", () => {
            const realm = "amido"
            expect(formatPrice(10000.5, 12, "CHF", "de-ch", realm)).toEqual("CHF 10.000,50 - CHF 12")
        })
        // Bug 34159
        it("should show 1000 delimiter as , for Switzerland EN", () => {
            const realm = "amido"
            expect(formatPrice(10000.5, 12, "CHF", "en-ch", realm)).toEqual("CHF 10,000.50 - CHF 12")
        })

        // Bug 34158
        it("should show a space correctly for Singapore", () => {
            const realm = "amido"
            expect(formatPrice(10000.5, 12, "SGD", "en-sg", realm)).toEqual("SGD 10,000.50 - SGD 12")
        })

        // Bug 34155
        it("should show a , correctly for Denmark thousand separator for English", () => {
            const realm = "amido"
            expect(formatPrice(10000.5, 12, "DKK", "en-dk", realm)).toEqual("DKK10,000.50 - DKK12")
        })
        it("should show a . correctly for Denmark thousand separator for Danish", () => {
            const realm = "amido"
            expect(formatPrice(10000.5, 12, "DKK", "da-dk", realm)).toEqual("DKK10.000,50 - DKK12")
        })

        // Bug 26597
        it("should show $ correctly for Mexico", () => {
            const realm = "amido"
            expect(formatPrice(10000.5, 12, "USD", "en-mx", realm)).toEqual("$10,000.50 - $12")
        })

    })
})
