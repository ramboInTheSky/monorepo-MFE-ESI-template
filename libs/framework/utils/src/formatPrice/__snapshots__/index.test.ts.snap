// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a price formatter When formatting the price with the config locale: ar-eg should render with no range  1`] = `"10 ج.م"`;

exports[`Given a price formatter When formatting the price with the config locale: ar-eg should render with range 1`] = `"10 ج.م - 15 ج.م"`;

exports[`Given a price formatter When formatting the price with the config locale: ar-lb should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: ar-lb should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: ar-om should render with no range  1`] = `"10 ر.ع"`;

exports[`Given a price formatter When formatting the price with the config locale: ar-om should render with range 1`] = `"10 ر.ع - 15 ر.ع"`;

exports[`Given a price formatter When formatting the price with the config locale: bg-bg should render with no range  1`] = `"BGN10"`;

exports[`Given a price formatter When formatting the price with the config locale: bg-bg should render with range 1`] = `"BGN10 - BGN15"`;

exports[`Given a price formatter When formatting the price with the config locale: cs-cz should render with no range  1`] = `"10 Kč"`;

exports[`Given a price formatter When formatting the price with the config locale: cs-cz should render with range 1`] = `"10 Kč - 15 Kč"`;

exports[`Given a price formatter When formatting the price with the config locale: da-dk should render with no range  1`] = `"DKK10"`;

exports[`Given a price formatter When formatting the price with the config locale: da-dk should render with range 1`] = `"DKK10 - DKK15"`;

exports[`Given a price formatter When formatting the price with the config locale: de-at should render with no range  1`] = `"10 €"`;

exports[`Given a price formatter When formatting the price with the config locale: de-at should render with range 1`] = `"10 € - 15 €"`;

exports[`Given a price formatter When formatting the price with the config locale: de-ch should render with no range  1`] = `"CHF 10"`;

exports[`Given a price formatter When formatting the price with the config locale: de-ch should render with range 1`] = `"CHF 10 - CHF 15"`;

exports[`Given a price formatter When formatting the price with the config locale: el-gr should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: el-gr should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-am should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-am should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ar should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ar should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-at should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-at should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-az should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-az should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-bg should render with no range  1`] = `"BGN 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-bg should render with range 1`] = `"BGN 10 - BGN 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-br should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-br should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-by should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-by should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ca should render with no range  1`] = `"CA$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ca should render with range 1`] = `"CA$10 - CA$15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ch should render with no range  1`] = `"CHF 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ch should render with range 1`] = `"CHF 10 - CHF 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-cl should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-cl should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-cy should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-cy should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-cz should render with no range  1`] = `"Kč10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-cz should render with range 1`] = `"Kč10 - Kč15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-dk should render with no range  1`] = `"DKK10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-dk should render with range 1`] = `"DKK10 - DKK15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ee should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ee should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-eg should render with no range  1`] = `"EGP10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-eg should render with range 1`] = `"EGP10 - EGP15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-es should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-es should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-fi should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-fi should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-fr should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-fr should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ge should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ge should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-gi should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-gi should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-gr should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-gr should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-hr should render with no range  1`] = `"HRK 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-hr should render with range 1`] = `"HRK 10 - HRK 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-hu should render with no range  1`] = `"10 Ft"`;

exports[`Given a price formatter When formatting the price with the config locale: en-hu should render with range 1`] = `"10 Ft - 15 Ft"`;

exports[`Given a price formatter When formatting the price with the config locale: en-id should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-id should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-in should render with no range  1`] = `"INR 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-in should render with range 1`] = `"INR 10 - INR 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-is should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-is should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-it should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-it should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-kr should render with no range  1`] = `"KRW10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-kr should render with range 1`] = `"KRW10 - KRW15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lb should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lb should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lt should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lt should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lu should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lu should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lv should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-lv should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-mx should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-mx should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-my should render with no range  1`] = `"MYR 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-my should render with range 1`] = `"MYR 10 - MYR 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-no should render with no range  1`] = `"kr10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-no should render with range 1`] = `"kr10 - kr15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-nz should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-nz should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-om should render with no range  1`] = `"OMR10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-om should render with range 1`] = `"OMR10 - OMR15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-pe should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-pe should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ph should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-ph should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-pk should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-pk should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-pt should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-pt should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-rs should render with no range  1`] = `"10 RSD"`;

exports[`Given a price formatter When formatting the price with the config locale: en-rs should render with range 1`] = `"10 RSD - 15 RSD"`;

exports[`Given a price formatter When formatting the price with the config locale: en-se should render with no range  1`] = `"kr10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-se should render with range 1`] = `"kr10 - kr15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-sg should render with no range  1`] = `"SGD 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-sg should render with range 1`] = `"SGD 10 - SGD 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-si should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-si should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-th should render with no range  1`] = `"$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-th should render with range 1`] = `"$10 - $15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-tr should render with no range  1`] = `"TRY 10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-tr should render with range 1`] = `"TRY 10 - TRY 15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-tw should render with no range  1`] = `"NT$10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-tw should render with range 1`] = `"NT$10 - NT$15"`;

exports[`Given a price formatter When formatting the price with the config locale: en-za should render with no range  1`] = `"R10"`;

exports[`Given a price formatter When formatting the price with the config locale: en-za should render with range 1`] = `"R10 - R15"`;

exports[`Given a price formatter When formatting the price with the config locale: es-es should render with no range  1`] = `"10 €"`;

exports[`Given a price formatter When formatting the price with the config locale: es-es should render with range 1`] = `"10 € - 15 €"`;

exports[`Given a price formatter When formatting the price with the config locale: fr-ca should render with no range  1`] = `"CA$10"`;

exports[`Given a price formatter When formatting the price with the config locale: fr-ca should render with range 1`] = `"CA$10 - CA$15"`;

exports[`Given a price formatter When formatting the price with the config locale: fr-fr should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: fr-fr should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: fr-lu should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: fr-lu should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: hu-hu should render with no range  1`] = `"10 Ft"`;

exports[`Given a price formatter When formatting the price with the config locale: hu-hu should render with range 1`] = `"10 Ft - 15 Ft"`;

exports[`Given a price formatter When formatting the price with the config locale: it-it should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: it-it should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: ko-kr should render with no range  1`] = `"KRW10"`;

exports[`Given a price formatter When formatting the price with the config locale: ko-kr should render with range 1`] = `"KRW10 - KRW15"`;

exports[`Given a price formatter When formatting the price with the config locale: ms-my should render with no range  1`] = `"MYR 10"`;

exports[`Given a price formatter When formatting the price with the config locale: ms-my should render with range 1`] = `"MYR 10 - MYR 15"`;

exports[`Given a price formatter When formatting the price with the config locale: ru-ee should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: ru-ee should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: ru-lt should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: ru-lt should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: ru-lv should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: ru-lv should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: sl-si should render with no range  1`] = `"€10"`;

exports[`Given a price formatter When formatting the price with the config locale: sl-si should render with range 1`] = `"€10 - €15"`;

exports[`Given a price formatter When formatting the price with the config locale: sv-se should render with no range  1`] = `"kr10"`;

exports[`Given a price formatter When formatting the price with the config locale: sv-se should render with range 1`] = `"kr10 - kr15"`;

exports[`Given a price formatter When formatting the price with the config locale: zh-tw should render with no range  1`] = `"NT$10"`;

exports[`Given a price formatter When formatting the price with the config locale: zh-tw should render with range 1`] = `"NT$10 - NT$15"`;
